CRM/Contact/BAO/Contact/Permission.php

```diff
@@ -43,8 +133,16 @@ class CRM_Contact_BAO_Contact_Permission {
    *   true if the user has permission, false otherwise
    */
   public static function allow($id, $type = CRM_Core_Permission::VIEW) {
-    $tables = array();
-    $whereTables = array();
+    // get logged in user
+    $session   = CRM_Core_Session::singleton();
+    $contactID = (int) $session->get('userID');
+
+    // first: check if contact is trying to view own contact
+    if ($type == CRM_Core_Permission::VIEW && CRM_Core_Permission::check('view my contact')
```

Comment:
I decided to not do that for 'view/edit _my_ contact', there might be scenarios where people should be able to e.g. set a flag to their contact without seeing all data....